### DISCUSSION TITLE
tasks of file metabolicTasks_full different from the same tasks in file metabolicTasks_essential

### DISCUSSION THREAD

**Discussion by PkiwiBird** [SysBioChalmers/Human-GEM#372] [General]
Some tasks present in both files 'metabolicTasks_full' and 'metabolicTasks_essential' files although with the same names are slightly different (see bellow). Which file is the correct one (should we use)?

Examples:
  ```
                                                                                metabolicTasks_essential                                         metabolicTasks_full
Aerobic rephosphorylation of ATP from glucose
column Out:                                                            H2O, CO2, H+                                                                H2O, CO2
ATP de novo synthesis                                                                                                                                           
column EQU:                                                           ATP[c] + H2O[c] ....                                                        no reaction

```

**Comment by JonathanRob** [comment]
Hi @PkiwiBird, these two task files were taken from different sources and therefore were developed separately. In general, we haven't attempted to reconcile their differences, but I think it would be a good thing to do.

The `metabolicTasks_essential` task file contains a list of tasks that all (human) cells should be able to perform to remain viable, regardless of cell type. However, the `metabolicTasks_full` task file is a longer list of tasks that includes some processes that may not be necessary for all cell types, so it's not necessarily expected that all models should be able to complete all of the tasks in the full list.

Feel free to edit the task files to get them in sync on the tasks that they share - I would recommend changing the `metabolicTasks_full` tasks to look like those in the essential file, but if any of the essential tasks look incorrect, then we can consider changing those instead.

**Reply by PkiwiBird** [reply]
Hi, checked now that some of those small inconsistencies are still there in the 'real files'. When I have time then I'll make the changes you suggested. thank you.

**Comment by PkiwiBird** [comment]
Hi, what is the _metabolicTasks_CellfieConsensus_ task file, and are those tasks supposed to work on the generic human1 model?

**Comment by PkiwiBird** [comment]
Also, wanted to ask, in `metabolicTasks_full.txt` when there is an `EQU` with empty LB and empty UB (e.g. glucose-1-phosphate[c] + H2O[c] = glucose[c] + Pi[c]), does it mean bounds are -1000.0 to 1000.0 or from 0.0 to 1000.0 ? 
Because, unlike metabolicTasks_essential.txt , there is no signal like this `=>` or `<=>` in `EQU` to indicate the direction of reaction, just an `=`

**Comment by JonathanRob** [comment]
> what is the metabolicTasks_CellfieConsensus task file, and are those tasks supposed to work on the generic human1 model?

Ah, this is the magic of GitHub - a lot of information can be found within the repository history/discussions/issues. For example, if you search on GitHub (within the Human-GEM repository) for "CellfieConsensus", you'll find Issue #280, which shows the thread where these tasks were discussed and introduced. Briefly, these tasks were developed/presented in a different study, but have been adapted to work with RAVEN. Though some of the tasks were found to fail for Human-GEM, which have not yet been addressed - something which would be great for someone to explore at some point!

> Also, wanted to ask, in `metabolicTasks_full.txt` when there is an `EQU` with empty LB and empty UB (e.g. glucose-1-phosphate[c] + H2O[c] = glucose[c] + Pi[c]), does it mean bounds are -1000.0 to 1000.0 or from 0.0 to 1000.0 ?
Because, unlike metabolicTasks_essential.txt , there is no signal like this `=>` or `<=>` in `EQU` to indicate the direction of reaction, just an `=`

The `EQU` field itself temporarily adds a reaction to the model for that task, where the reaction arrow (`=>` or `<=>`) indicates its reversibility (with default bounds of 0 to 1000, or -1000 to 1000, respectively). These default bounds can be changed if a value is included in the corresponding `EQU LB` or `EQU UB` column.

I do not see any case in `metabolicTasks_Essential.txt` where the reaction symbol is just `=` without arrows - where exactly do you see this?

**Comment by PkiwiBird** [comment]
> The `EQU` field itself temporarily adds a reaction to the model for that task, where the reaction arrow (`=>` or `<=>`) indicates its reversibility (with default bounds of 0 to 1000, or -1000 to 1000, respectively). These default bounds can be changed if a value is included in the corresponding `EQU LB` or `EQU UB` column.
> 
> I do not see any case in `metabolicTasks_Essential.txt` where the reaction symbol is just `=` without arrows - where exactly do you see this?
>
It's in EQU column of the file `metabolicTasks_full.txt`. In tasks full, empty LB and empty UB (e.g. glucose-1-phosphate[c] + H2O[c] = glucose[c] + Pi[c]), does it mean bounds are -1000.0 to 1000.0 or from 0.0 to 1000.0 ?

**Comment by PkiwiBird** [comment]
> > what is the metabolicTasks_CellfieConsensus task file, and are those tasks supposed to work on the generic human1 model?
> 
> Ah, this is the magic of GitHub - a lot of information can be found within the repository history/discussions/issues. For example, if you search on GitHub (within the Human-GEM repository) for "CellfieConsensus", you'll find Issue #280, which shows the thread where these tasks were discussed and introduced. Briefly, these tasks were developed/presented in a different study, but have been adapted to work with RAVEN. Though some of the tasks were found to fail for Human-GEM, which have not yet been addressed - something which would be great for someone to explore at some point!
>
Ok, thank you.

**Comment by JonathanRob** [comment]
> It's in EQU column of the file `metabolicTasks_full.txt`. In tasks full, empty LB and empty UB (e.g. glucose-1-phosphate[c] + H2O[c] = glucose[c] + Pi[c]), does it mean bounds are -1000.0 to 1000.0 or from 0.0 to 1000.0 ?

I'm sorry, I still don't find this in [`metabolicTasks_full.txt`](https://github.com/SysBioChalmers/Human-GEM/blob/main/data/metabolicTasks/metabolicTasks_Full.txt) either - all the equations I see in that document contain arrows. What branch/version of Human-GEM are you looking at, and what is the exact task in which you see this `=` without an arrow?

**Comment by PkiwiBird** [comment]
> > It's in EQU column of the file `metabolicTasks_full.txt`. In tasks full, empty LB and empty UB (e.g. glucose-1-phosphate[c] + H2O[c] = glucose[c] + Pi[c]), does it mean bounds are -1000.0 to 1000.0 or from 0.0 to 1000.0 ?
> 
> I'm sorry, I still don't find this in [`metabolicTasks_full.txt`](https://github.com/SysBioChalmers/Human-GEM/blob/main/data/metabolicTasks/metabolicTasks_Full.txt) either - all the equations I see in that document contain arrows. What branch/version of Human-GEM are you looking at, and what is the exact task in which you see this `=` without an arrow?

I'm sorry, you're right. I was looking inside another file with the same name on my computer.

**Comment by mihai-sysbio** [comment]
I'm wondering if there is any action to be taken here, or if it would make sense to convert the current issue into a _Discussion_.

**Comment by mihai-sysbio** [comment]
I'm wondering if there is any action to be taken here, or if it would make sense to convert the current issue into a _Discussion_.

**Comment by JonathanRob** [comment]
@mihai-sysbio yes, I think at this stage it fits better as a Discussion.
